hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/commit/magic/MagicS3GuardCommitter.java

```diff
@@ -131,16 +134,19 @@ protected ActiveCommit listPendingUploadsToCommit(
    * Delete the magic directory.
    */
   public void cleanupStagingDirs() {
-    final Path out = getOutputPath();
-    Path path = magicSubdir(out);
-    try(DurationInfo ignored = new DurationInfo(LOG, true,
-        "Deleting magic directory %s", path)) {
-      Invoker.ignoreIOExceptions(LOG, "cleanup magic directory", path.toString(),
-          () -> deleteWithWarning(getDestFS(), path, true));
-      // and the job temp directory with manifests
-      Invoker.ignoreIOExceptions(LOG, "cleanup job directory", path.toString(),
-          () -> deleteWithWarning(getDestFS(),
-              new Path(out, TEMP_DATA), true));
+    if (cleanupMagicDirectory) {
+      final Path out = getOutputPath();
+      Path path = getMagicJobPath(getUUID(), out);
+      try (DurationInfo ignored = new DurationInfo(LOG, true,
+          "Deleting magic directory %s", path)) {
+        Invoker.ignoreIOExceptions(LOG, "cleanup magic directory", path.toString(),
+            () -> deleteWithWarning(getDestFS(), path, true));
+        // and the job temp directory with manifests
+        Invoker.ignoreIOExceptions(LOG, "cleanup job directory", path.toString(),
```

Comment:
As per this change, The metadata (.pending set) of all the pending commits will be stored in /__magic_job-<id>. So there is no common root metadata directory unlike the current case where it is stored in __magic/job-<id>.

This change was introduced to fix https://issues.apache.org/jira/browse/HADOOP-18568, Where jobs were bottlenecked by the time to delete the __magic/job-<id>/ directory and in this case it will be __magic_job-<id>.

Reply:
Just to keep the scope of this change confined. I will remove this change which can be handled in the original Jira itself. (If it needs more discussion)